### ISSUE TITLE
[Bug] Typescript code in Vue template is not formatted

### ISSUE THREAD

**Issue by wenfangdu** [prettier/prettier#14432]
**Prettier 2.8.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAzmATgSwA4wAEANgIZQDmAvADogzp0B8qA9JrgUzVD6vAFs8ZeNyiFCqSEMJIAZhAi0QAClJJ0MXJQCUVJgEY6rMW0HDSoniAA0ICARzR0yUKSxYIAdwAK7hC4opMRepACeLnYARlikYADWcDAAyqQCcAAyOFBwyHLB6HDRsQlJyXhx2RTIWgCuRSBwAlFwACatbRnkFLWkFHAAYhBYApYwVcggpLUwELYgABYwAsQA6gs48OgVYHDJAZs4AG6bYZNg6JEg2YVYMD6xFKN5BQ0AVugAHslVxHAAirUIPAXsRCnYKlhbpMjvV5nhtDBVjhWjAFsgABwABghnkKq1ieEmCLgtyOuTsAEcgfAHg5AlN0ABaHJtNrzLBwak4TkPPrPeSvOyFAQ4GpYerC35wACCMC0OCiMzgPjgWCyOVB4JA6GlgOBuUFYIaMFIUWRqPRSAATHYtKQcMQqgBhCACAWNdAAVnmtUKABUzYF8sa7LC4ABJKAdWDJbD4GAy6PJGBhP5auAAX0zQA)

```sh
--parser vue
```

**Input:**

```vue
<script lang="ts"></script>

<template>
  <comp :foo="(a:string)=>1"/>
</template>
```

**Output:**

```vue
<script lang="ts"></script>

<template>
  <comp :foo="(a:string)=>1" />
</template>
```

**Expected behavior:**

```vue
<script lang="ts"></script>

<template>
  <comp :foo="(a: string) => 1" />
</template>
```

**Comment by fisker** [comment]
We only support typescript syntax when there is scripts with `[lang="typescript"]`, but this still a bug. Updated code example to add a `<script>` block.

**Comment by seiyab** [comment]
@wenfangdu 
Would you try the playground of https://github.com/prettier/prettier/pull/14506 ?
I'm not familiar with Vue enough.

**Comment by wenfangdu** [comment]
@seiyab Thanks, it seems to be working:
![image](https://user-images.githubusercontent.com/28700378/225190668-c214feff-aad7-4674-95ab-30544cdca26c.png)

**Comment by sosukesuzuki** [comment]
Fixed by #14587
